### ISSUE TITLE
Fails to process kafka-streams-avro-serde-7.3.1.jar

### ISSUE THREAD

**Issue by vspiliop** [reallyinsane/mathan-latex-maven-plugin#7]
Plugin fails if a maven dependency has the following structure:

```
META-INF/MANIFEST.MF
META-INF/
```

if the iterator return MANIFEST.MF first and META-INF/ second. As the META-INF/ will be already created as part of the processing of MANIFEST.MF.
 

`Caused by: java.io.IOException: Could not create directory /var/folders/gr/0k9ly3t15c11g26_jl9sx3km0000gn/T/8cbe4bc9-9e90-438e-a967-fccb253b7f3a/META-INF
    at io.mathan.latex.core.Utils.extractArchive (Utils.java:150)
    at io.mathan.maven.latex.internal.MavenBuild.extractArchive (MavenBuild.java:161)
    at io.mathan.maven.latex.internal.MavenBuild.resolveDependency (MavenBuild.java:149)
    at io.mathan.maven.latex.internal.MavenBuild.resolveDependencies (MavenBuild.java:86)
    at io.mathan.latex.core.MathanLatexRunner.executeSteps (MathanLatexRunner.java:100)
    at io.mathan.latex.core.MathanLatexRunner.execute (MathanLatexRunner.java:78)
    at io.mathan.maven.latex.MathanLatexMojo.execute (MathanLatexMojo.java:162)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)`

You can try to process `io/confluent/kafka-streams-avro-serde/7.3.1/kafka-streams-avro-serde-7.3.1.jar` for reproducing the issue.

**Comment by vspiliop** [comment]
Fix could be:

```
if (!directory.exists() && !directory.mkdirs()) {
            throw new IOException("Could not create directory " + directory.getAbsolutePath());
          }
```

at `Utils.extractArchive()`

**Comment by reallyinsane** [comment]
Should be working with the snapshot. Can you have a look?

**Comment by vspiliop** [comment]
Looks great! thanks!

**Comment by vspiliop** [comment]
Hello @reallyinsane! Are you planning to release soon?

**Comment by reallyinsane** [comment]
> Hello @reallyinsane! Are you planning to release soon?

I think in April

**Comment by reallyinsane** [comment]
Resolved with release 1.0.4
